src/nepiada/utils/graphs.py

```diff
@@ -59,8 +74,68 @@ def update_graphs(self, agents):
             # Do not update the observation graph, because it has been configured to be static
             pass
 
-        # Update communication graph here
-        # Nothing to do, as we have defined communication graph to be static
+    def _update_comm_graph(self, agents):
+        """
+        If dynamic communication graph is enabled, this function will update the communication graph based on the current positions of the agents. 
+        It will also enforce a minimum number of agents that can communicate with each other, if the number of agents that can communicate with each other
+        is less than the minimum specified by the configuration.
+        If static communication graph, this function will do nothing since the communication graph is already configured to allow for full communication.
+    
+        params:
+            agents: A dictionary of agents, where the key is the agent's name and the value is the agent object.
+        """
+        if self.dynamic_comms:
+            # Reset the graph
+            self.comm = {agent: [] for agent in agents}
+            for agent_name, agent in agents.items():
+                x_coord = agent.p_pos[0]
+                y_coord = agent.p_pos[1]
+
+                distances_to_agents_not_added = []
+                # Update communication graph here
+                for other_agent_name, other_agent in agents.items():
+                    if other_agent_name != agent_name:
+                        # Calculate the distance between them
+                        delta_x = other_agent.p_pos[0] - x_coord
+                        delta_y = other_agent.p_pos[1] - y_coord
+
+                        distance = np.sqrt(delta_x**2 + delta_y**2)
+
+                        # If within communication radius add them to the graph
+                        # If not within the communication radius, add them to a list of agents that are not added,
+                        # for potential future addition based on enforcing minimum number of comm agents.
+                        if distance < self.dynamic_comms_radius:
```

Comment:
A small optimization here would be to break the for loop once we have appended the enforced_minimum number of agents, to avoid looping over all others.

Reply:
We wouldn't want to break this loop right? Even if we hit the minimum if theres more agents within the radius then we want to add them.

Reply:
Oh yes, you have a point. We shouldn't break it then